RecoEgamma/ElectronIdentification/plugins/ElectronMVANtuplizer.cc

```diff
@@ -0,0 +1,534 @@
+
+
+      // gap variables
+      bool eleIsEB_;
+      bool eleIsEE_;
+      bool eleIsEBEtaGap_;
+      bool eleIsEBPhiGap_;
+      bool eleIsEBEEGap_;
+      bool eleIsEEDeeGap_;
+      bool eleIsEERingGap_;
+
+      // to hold ID decisions and categories
+      std::vector<int> mvaPasses_;
+      std::vector<float> mvaValues_;
+      std::vector<int> mvaCats_;
+
+      // config
+      const bool isMC_;
+      const double deltaR_;
+      //const bool saveIDVariables_;
+      const double ptThreshold_;
+
+      // ID decisions objects
+      const std::vector< std::string > eleMapTags_;
+      std::vector< edm::EDGetTokenT< edm::ValueMap<bool> > > eleMapTokens_;
+      const std::vector< std::string > eleMapBranchNames_;
+      const size_t nEleMaps_;
+
+      // MVA values and categories (optional)
+      const std::vector< std::string > valMapTags_;
+      std::vector< edm::EDGetTokenT<edm::ValueMap<float> > > valMapTokens_;
+      const std::vector< std::string > valMapBranchNames_;
+      const size_t nValMaps_;
+
+      const std::vector< std::string > mvaCatTags_;
+      std::vector< edm::EDGetTokenT<edm::ValueMap<int> > > mvaCatTokens_;
+      const std::vector< std::string > mvaCatBranchNames_;
+      const size_t nCats_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+enum ElectronMatchType {
+                        UNMATCHED,
+                        TRUE_PROMPT_ELECTRON,
+                        TRUE_ELECTRON_FROM_TAU,
+                        TRUE_NON_PROMPT_ELECTRON,
+                       }; // The last does not include tau parents
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+ElectronMVANtuplizer::ElectronMVANtuplizer(const edm::ParameterSet& iConfig)
+ :
+  src_                   (consumes<edm::View<reco::GsfElectron> >(iConfig.getParameter<edm::InputTag>("src"))),
+  vertices_              (consumes<std::vector<reco::Vertex> >(iConfig.getParameter<edm::InputTag>("vertices"))),
+  pileup_                (consumes<std::vector< PileupSummaryInfo > >(iConfig.getParameter<edm::InputTag>("pileup"))),
+  genParticles_          (consumes<edm::View<reco::GenParticle> >(iConfig.getParameter<edm::InputTag>("genParticles"))),
+  srcMiniAOD_            (consumes<edm::View<reco::GsfElectron> >(iConfig.getParameter<edm::InputTag>("srcMiniAOD"))),
+  verticesMiniAOD_       (consumes<std::vector<reco::Vertex> >(iConfig.getParameter<edm::InputTag>("verticesMiniAOD"))),
+  pileupMiniAOD_         (consumes<std::vector< PileupSummaryInfo > >(iConfig.getParameter<edm::InputTag>("pileupMiniAOD"))),
+  genParticlesMiniAOD_   (consumes<edm::View<reco::GenParticle> >(iConfig.getParameter<edm::InputTag>("genParticlesMiniAOD"))),
+  mvaVarMngr_            (iConfig.getParameter<std::string>("variableDefinition")),
+  isMC_                  (iConfig.getParameter<bool>("isMC")),
+  deltaR_                (iConfig.existsAs<double>("deltaR")          ? iConfig.getParameter<double>("deltaR"): 0.1),
```

Comment:
Better to use the fillDescription() method (which is already setup for this class) and remove the need of all ```iConfig.existsAs```